### PR TITLE
stm32: nucleo_g474re: reflash board successively with pyocd

### LINKED ISSUE #23351: boards: nucle_g474re: west flash doesn't work
**Describe the bug**
Flashing of the board doesn't work.
It shows successful, but the board does not blink.
**To Reproduce**
1. west build -b nucleo_g474re zephyr/samples/basic/blinky
2. west flash

**Expected behavior**
LED should blink.

**Impact**
Can't flash the board

**Environment (please complete the following information):**
 - OS: Linux
 - pyocd

### PR DESCRIPTION
**Pull request by FRASTM**
It happens that the nucleo_g474re board  cannot be flashed a second time and reboot with pyocd.
(copy binary to the target board is still functional)
This patch changes the board_runner_args(pyocd "--target=stm32g474rbtx") in the board.cmake 
Use west flash to program the target board /zephyr/boards/arm/nucleo_g474re

Signed-off-by: Francois Ramu francois.ramu@st.com

Fixes #23351

### CODE REVIEW

File: boards/arm/nucleo_g474re/board.cmake [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # SPDX-License-Identifier: Apache-2.0
-board_runner_args(pyocd "--target=stm32g474retx")
+board_runner_args(pyocd "--target=stm32g474rbtx")
```
Comment (alexanderwachter) [inline boards/arm/nucleo_g474re/board.cmake]: This change works for me! Thanks

**Review by alexanderwachter — APPROVED** [review]

File: boards/arm/nucleo_g474re/board.cmake (hunk 1 above)
Comment (erwango) [inline boards/arm/nucleo_g474re/board.cmake]: This would deserve a comment so we know this is made on purpose, to solve a flashing issue.
Also, are we sure this has no impact otherwise (which should be advertised as limitations )?

File: soc/arm/st_stm32/stm32g4/Kconfig.defconfig.stm32g474re [hunk 2]
```diff
@@ -6,11 +6,9 @@
 if SOC_STM32G474XX
 
 config SOC
-	string
 	default "stm32g474xx"
 
 config NUM_IRQS
-	int
 	default 102
```
Comment (erwango) [inline soc/arm/st_stm32/stm32g4/Kconfig.defconfig.stm32g474re]: Let's split this is a dedicated commit (same PR could be used).

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by FRASTM** [comment]
separate in 2 commits

File: soc/arm/st_stm32/stm32g4/Kconfig.defconfig.stm32g474re (hunk 2 above)
Comment (FRASTM) [reply soc/arm/st_stm32/stm32g4/Kconfig.defconfig.stm32g474re]: done into 13c405dda1e2f443a592a5464120fdf04cddbf84 and  3611b5da65c826dd2d62ac3d2517f316a0670040

File: boards/arm/nucleo_g474re/board.cmake (hunk 1 above)
Comment (FRASTM) [reply boards/arm/nucleo_g474re/board.cmake]: adding a comment

Comment (FRASTM) [reply boards/arm/nucleo_g474re/board.cmake]: adding a comment

**Comment by FRASTM** [comment]
adding a comment in boards/arm/nucleo_g474re/board.cmake

**Review by erwango — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
